### PR TITLE
Disable warnings for input global variables

### PR DESCRIPTION
**Pull request by venkataram-nv**
Fixes an undesired warning issued when using GLSL layout variables. The higher level problem is that the warning system was not skipping checks for input global variables.

### CODE REVIEW

File: tests/diagnostics/uninitialized-globals.slang [hunk 1]
```diff
@@ -45,4 +45,34 @@ float use_later_writte()
     return writtenLater;
 }
 
+// Should not warn here (glsl layout decoration check)
```
Comment (ArielG-NV) [inline tests/diagnostics/uninitialized-globals.slang:48]: I don't think this comment is quite specific enough with "(glsl layout decoration check)".
```suggestion
// Varying inputs never warn
```

File: tests/diagnostics/uninitialized-globals.slang [hunk 2]
```diff
@@ -45,4 +45,34 @@ float use_later_writte()
...
+layout(location = 0) in vec4 data;
+
+vec4 glsl_layout_in_ok()
+{
+    return data;
+}
+
+// Invalid use of layout variables
```
Comment (ArielG-NV) [inline tests/diagnostics/uninitialized-globals.slang:56]: I don't think this comment describes why we have this test case.

I suggest something like the following:
```suggestion
// Varying outputs should be initialized by the end of a shader.
```

File: tests/diagnostics/uninitialized-globals.slang [hunk 3]
```diff
@@ -45,4 +45,34 @@ float use_later_writte()
...
+layout(location = 1) out vec4 x;
+
+vec4 glsl_layout_out_undefined()
+{
+    //CHK-DAG: ([[# @LINE + 1]]): warning 41017: use of uninitialized global variable 'x'
+    return x;
+}
+
+// OK use of layout variables
```
Comment (ArielG-NV) [inline tests/diagnostics/uninitialized-globals.slang]: I don't think this comment is needed if we explain why we warn (programmer should see that the variable is initialized).

**Review by jkwak-work — APPROVED** [review]
Looks good to me

**Review by csyonghe — APPROVED** [review]

**Review by jkwak-work — APPROVED** [review]

**Pull request merged**